cpp/tests/io/parquet_test.cpp

```diff
@@ -4215,4 +4268,194 @@ TEST_F(ParquetReaderTest, EmptyColumnsParam)
+     {'F', 'r', 'i', 'd', 'a', 'y'}},
+    {{'M', 'o', 'n', 'd', 'a', 'y'}, {'F', 'r', 'i', 'd', 'a', 'y'}}};
+
+  std::vector<std::unique_ptr<column>> cols;
+  cols.push_back(int_col.release());
+  cols.push_back(float_col.release());
+  cols.push_back(list_list_int_col.release());
+  auto expected = std::make_unique<table>(std::move(cols));
+  EXPECT_EQ(3, expected->num_columns());
+  cudf_io::table_input_metadata ouput_metadata(*expected);
+  ouput_metadata.column_metadata[0].set_name("col_other");
+  ouput_metadata.column_metadata[1].set_name("col_float");
+  ouput_metadata.column_metadata[2].set_name("col_binary").set_output_as_binary(true);
+
+  auto filepath = temp_env->get_temp_filepath("NestedByteArray.parquet");
+  cudf_io::parquet_writer_options out_opts =
+    cudf_io::parquet_writer_options::builder(cudf_io::sink_info{filepath}, expected->view())
+      .metadata(&ouput_metadata);
+  cudf_io::write_parquet(out_opts);
+
+  cudf_io::parquet_reader_options in_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath})
+      .convert_binary_to_strings({true, true, true, true, true, true, true, true});
+  auto result = cudf_io::read_parquet(in_opts);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected->view(), result.tbl->view());
+}
+
+TEST_F(ParquetWriterTest, ByteArrayStats)
+{
+  // check that byte array min and max statistics are written as expected. If a byte array is
+  // written as a string, max utf8 is 0xf7bfbfbf and so the minimum value will be set to that value
+  // instead of a potential minimum higher than that.
+  std::vector<uint8_t> expected_col0_min{0xf0};
+  std::vector<uint8_t> expected_col0_max{0xf0, 0xf5, 0xf5};
+  std::vector<uint8_t> expected_col1_min{0xfe, 0xfe, 0xfe};
+  std::vector<uint8_t> expected_col1_max{0xfe, 0xfe, 0xfe};
+
+  cudf::test::lists_column_wrapper<uint8_t> list_int_col0{
+    {0xf0}, {0xf0, 0xf5, 0xf3}, {0xf0, 0xf5, 0xf5}};
+  cudf::test::lists_column_wrapper<uint8_t> list_int_col1{
+    {0xfe, 0xfe, 0xfe}, {0xfe, 0xfe, 0xfe}, {0xfe, 0xfe, 0xfe}};
+
+  std::vector<std::unique_ptr<column>> cols;
+  cols.push_back(list_int_col0.release());
+  cols.push_back(list_int_col1.release());
+  auto expected = std::make_unique<table>(std::move(cols));
+  EXPECT_EQ(2, expected->num_columns());
+  cudf_io::table_input_metadata ouput_metadata(*expected);
+  ouput_metadata.column_metadata[0].set_name("col_binary0").set_output_as_binary(true);
+  ouput_metadata.column_metadata[1].set_name("col_binary1").set_output_as_binary(true);
+
+  auto filepath = temp_env->get_temp_filepath("ByteArrayStats.parquet");
+  cudf_io::parquet_writer_options out_opts =
+    cudf_io::parquet_writer_options::builder(cudf_io::sink_info{filepath}, expected->view())
+      .metadata(&ouput_metadata);
+  cudf_io::write_parquet(out_opts);
+
+  cudf_io::parquet_reader_options in_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath})
+      .convert_binary_to_strings({true, true});
```

Comment:
Surprised C++ didn't try to interpret this is `std::vector<?>(count, value)`. It's been a pain for some other tests with two element vectors.

Reply:
In my limited experience (and understanding), with list initialization, `std::vector::vector(count, val)` isn't called.
I honestly thought the advice ran the other way: when trying to call `vector(count, val)`, we shouldn't be using `{count, val}`.

Reply:
Looks like I mixed it up with some other case. But there were definitely some tests when constructor calls were ambiguous when passing exactly two elements. Either way, not relevant to the task at hand :)

Reply:
I think Github lost my reply to this: 

I honestly thought that things worked in the other direction: that when list initialization is used, `std::vector::vector(count, value)` isn't called.

I thought the advice was that when one intends to call `vector(count, value)`, we should explicitly use `(count, value)` rather than `{count, value}`. :man_shrugging:

Reply:
(I despise the delay in viewing Github comments, including my own.)